src/home/fetch-github/fetch-data.ts

```diff
@@ -0,0 +1,187 @@
+import { Octokit } from "@octokit/rest";
+import { GitHubAggregated, GitHubIssue, GitHubNotification, GitHubNotifications, GitHubPullRequest } from "../github-types";
+import { getGitHubAccessToken } from "../getters/get-github-access-token";
+import { handleRateLimit } from "./handle-rate-limit";
+import { RequestError } from "@octokit/request-error";
+
+export const organizationImageCache = new Map<string, Blob | null>(); // this should be declared in image related script
+
+// Generalized function to fetch notifications from GitHub
+async function fetchNotifications(): Promise<GitHubNotifications | null> {
+  const providerToken = await getGitHubAccessToken();
+  const octokit = new Octokit({ auth: providerToken });
+
+  try {
+    const notifications = (await octokit.request("GET /notifications")).data as GitHubNotifications;
+    console.log("unfiltered", notifications);
+    return notifications;
+  } catch (error) {
+    if (error instanceof RequestError && error.status === 403) {
+      await handleRateLimit(octokit, error);
+    }
+    console.warn("Error fetching notifications:", error);
+  }
+  return null;
+}
+
+// Pre-filter notifications by general rules (repo filtering and ignoring CI activity)
+function preFilterNotifications(notifications: GitHubNotification[]): GitHubNotifications {
+  return notifications.filter((notification) => {
+    // Ignore based on reason
+    if (
+      ["comment", "ci_activity", "invitation", "member_feature_requested", "security_advisory_credit", "state_change", "team_mention"].includes(
+        notification.reason
+      )
+    ) {
+      return false;
+    }
+
+    // Ignore notifications from orgs that are not relevant
+    const repoName = notification.repository.full_name.split("/")[0];
+    return ["ubiquity", "ubiquity-os", "ubiquity-os-marketplace"].includes(repoName);
```

Comment:
This is wrong to hard code org names like this

Reply:
do you want to pull from a config file?

Reply:
It should be every org on the directory